docs/language/informal/toplevel-inference.md

```diff
@@ -0,0 +1,360 @@
+  * Inference dependencies are those of the operand.
+* A list literal or a map literal with explicit type arguments has the type
+   `List<T>` or `Map<K, V>` respectively, where `<T>` or `<K, V>` are the
+   provided type arguments.
+   * No inference dependencies
+* A non-empty list literal with no type arguments, and for which all of the
+   elements are IEs, has the type `List<T>` where `T` is the least upper bound
+   of the inferred types of the elements.
+   * If the elements do not all have an inferred type, it is an error
+   * The inference dependencies of the list literal are the collected
+      dependencies of the elements.
+* An non-const empty list literal with no type arguments has the type
+  `List<dynamic>`.
+   * No inference dependencies.
+* A const empty list literal with no type arguments has the type
+  `List<Null>`.
+   * No inference dependencies.
+* A map literal with no type arguments, and for which all of the keys and
+   values are IEs, has the type `Map<K, V>` where `K` is the least upper bound of
+   the inferred types of the provided keys, and `V` is the least upper bound of
+   the provided values.
+   * If the keys and values do not all have an inferred type, it is an error.
+   * The inference dependencies of the map literal are the collected
+      dependencies of the keys and values.
+* An non-const empty map literal with no type arguments has the type
+  `Map<dynamic, dynamic>`.
+   * No inference dependencies.
+* A const empty map literal with no type arguments has the type `Map<Null,
+  Null>`.
+   * No inference dependencies.
+* A function literal with an expression body for which all parameter types are
+   type annotated and for which the return expression is an IE and has an
+   inferred type, has the corresponding function type.
+   * If the return expression has no inferred type it is an error.
+   * The inference dependencies are the inference dependencies of the return
+   expression.
+   * If the function is marked `async`, then the return type of the function is
+     inferred as `Future<flatten<T>>` where `T` is the inferred type of the
+     return expression and flatten is as described in the spec.
+* An instance creation expression with no omitted type arguments has the
+   obvious type.
+   * No inference dependencies
+* An `as` expression has the cast to type
+   * No inference dependencies
+* An `is` expression has type boolean
+   * No inference dependencies
+* A simple or qualified identifier referring to a top level function, static
+   variable, field, getter; or a static class variable, static getter or method;
+   or an instance method; has the inferred type of the identifier.
+   * Otherwise, if the identifier has no inferred or annotated type then it is
+   an error.
+   * Note: specifically, references to instance fields and instance getters are
+     disallowed here.
+   * The inference dependency of the identifier is itself if the identifier is
```

Comment:
Consider changing this to `The inference dependency of the identifier is the referent if the referent is a candidate for inference...`